### PR TITLE
EvmTool Documentation

### PR DESCRIPTION
**Pull request by shemnon**
**Before pushing any commit in your pull request, make sure that you:**

- [X] read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [X] [signed all commits of for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [X] have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review. -->

## Describe the change

Initial set of documentation for Besu EVMTool

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

### For content changes

- [x] Doc content
- [x] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] Github integration

## Testing

See https://hyperledger-besu--495.org.readthedocs.build/en/495/Reference/Evm-Tool/

### CODE REVIEW

**Comment by MadelineMurray** [comment]
@shemnon - this looks great, thanks for putting it together.  I'm going to add a How To topic and move some of the content there.  From our current top level topics, How To -> Trouble Shoot -> Use EVM Tool looks like the best option but benchmarking isn't really troubleshooting.  Alternatively I could add a new category and have How To -> Benchmark and fuzz test -> Use the EVM Tool.  Have you got a preference?

**Comment by shemnon** [comment]
How to sounds fine.  I expect it to evolve possibly into a stand alone transaciton validator if the witness spec finalizes, but we can re-position if things actually happen.  "How To -> Trouble Shoot -> Use EVM Tool" would be my preference.

**Comment by holiman** [comment]
Awesome! I have a few questions, not sure where it's appropriate to file it, so I'll do it here for now. 

### stateroot

```
user@dockervm:~$ docker run -v /home/user/QubesIncoming/work/:/work  hyperledger/besu-evmtool:develop --json state-test /work/statetest1.json
{"pc":0,"op":"0x5b","gas":"0x2d1cc4","gasCost":"0x1","memSize":0,"stack":[],"depth":1,"error":null,"opName":"JUMPDEST"}
{"pc":1,"op":"0x3d","gas":"0x2d1cc3","gasCost":"0x2","memSize":0,"stack":[],"depth":1,"error":null,"opName":"RETURNDATASIZE"}
{"pc":2,"op":"0x30","gas":"0x2d1cc1","gasCost":"0x2","memSize":0,"stack":["0x0"],"depth":1,"error":null,"opName":"ADDRESS"}
{"pc":3,"op":"0x79","gas":"0x2d1cbf","gasCost":"0x3","memSize":0,"stack":["0xc94f5374fce5edbc8e2a8697c15331677e6ebf0b","0x0"],"depth":1,"error":null,"opName":"PUSH26"}
{"pc":30,"op":"0x68","gas":"0x2d1cbc","gasCost":"0x3","memSize":0,"stack":["0xc940b5f2046740058558468f238b85db7f6bbe3f3d51e92a3e32","0xc94f5374fce5edbc8e2a8697c15331677e6ebf0b","0x0"],"depth":1,"error":null,"opName":"PUSH9"}
{"pc":40,"op":"0x53","gas":"0x2d1cb9","gasCost":"0x7fffffffffffffff","memSize":0,"stack":["0xb7f7c4147541c695f3","0xc940b5f2046740058558468f238b85db7f6bbe3f3d51e92a3e32","0xc94f5374fce5edbc8e2a8697c15331677e6ebf0b","0x0"],"depth":1,"error":"Out of gas","opName":"MSTORE8"}
{"output":"","gasUsed":"0x2dc6c0","time":40081103,"test":"randomStatetestmartin-Wed_10_02_29-14338-0","fork":"Byzantium","d":0,"g":0,"v":0,"postHash":false,"postLogs":true}
```
How can I see the resulting post-stateroot ? 

### Additional (optional fields)

There are a few more json-fields that are implemented by geth, which would be neat if supported: 
-`rStack` - the return stack, 
- `refund` - the global refund counter value
- `returnData` - the context-current value of returndata

**Comment by shemnon** [comment]
Can you post the above to https://github.com/hyperledger/besu/pull/1320?  This issue is just the docs.

**Comment by shemnon** [comment]
@holiman Can you post the above to hyperledger/besu#1320? This issue is just the docs.

File: docs/Reference/Evm-Tool.md [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+
+    ```bash
+    --block-number=10000000
+    ```
+
+The block number to evaluate the code against.  This is useful to ensure that the EVM is evaluating the code against the correct fork, or to specify the specific world state when running with `rocksdb`.
+
+## EVM Tool State Test Options
+
+EVM Tool has a `stat-test` sub command that allows Ethereum State Tests to be evaluated.  Most of the options from EVM execution do not apply.
```
Comment (MariusVanDerWijden) [inline docs/Reference/Evm-Tool.md:317]:
```suggestion
EVM Tool has a `state-test` sub command that allows Ethereum State Tests to be evaluated.  Most of the options from EVM execution do not apply.
```

File: docs/Reference/Evm-Tool.md [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+    --receiver=<address>
+    ```
+
+=== "Example"
+
+    ```bash
+    --receiver=0x588108d3eab34e94484d7cda5a1d31804ca96fe7
+    ```
+
+The account the invocation will be sent from.  This account does not need to exist.
```
Comment (MadelineMurray) [inline docs/Reference/Evm-Tool.md:134]: @shemnon - confirming that `from` should be `to` here?

**Comment by shemnon** [comment]
Superseded by #502?

**Comment by MadelineMurray** [comment]
And then superseded by #509

**Pull request closed without merging**